packages/api-client/src/client/index.ts

```diff
@@ -127,6 +136,7 @@ export class ApiClient {
         // Send request
         const { fetch } = config;
         const fetchResponse = await fetch(request);
+        if (!options?.hideDeprecatedWarning) console.warn(`@opendatasoft/api-client : ${fetchResponse.headers.get('Ods-Explore-Api-Deprecation')}`);
```

Comment:
Shouldn't it only appear if there is a content in the header? Right now it runs even if the header is null.

Also, shouldn't we only run this once, to avoid spamming the console? We could store a `deprecationWarned` internal boolean var in the ApiClient class for that. In effect it would run once per instantation of the client, so once per page, once per run of a script etc.

Reply:
Ah yes, it's better indeed!